ktor-server/ktor-server-plugins/ktor-server-di/jvm/test/io/ktor/server/plugins/di/DependencyInjectionJvmTest.kt

```diff
@@ -62,48 +60,42 @@ class DependencyInjectionJvmTest {
     }
 
     @Test
-    fun `circular references from create`() = testApplication {
-        application {
-            assertFailsWith<CircularDependencyException> {
-                dependencies {
-                    provide<WorkExperience> { WorkExperience(this.resolve()) }
-                    provide<PaidWork> { PaidWork(this.resolve()) }
-                    provide<List<PaidWork>> { listOf(this.resolve()) }
-                }
-                val workExperience: WorkExperience = dependencies.create()
-                fail("This should fail but returned $workExperience")
+    fun `circular references from create`() = runTestDI {
+        assertFailsWith<CircularDependencyException> {
+            dependencies {
+                provide<WorkExperience> { WorkExperience(this.resolve()) }
+                provide<PaidWork> { PaidWork(this.resolve()) }
+                provide<List<PaidWork>> { listOf(this.resolve()) }
             }
+            val workExperience: WorkExperience = dependencies.create()
+            fail("This should fail but returned $workExperience")
         }
```

Comment:
Minor. We could wrap only `dependencies.create` with `assertFailWith` to make it clear that the failure occurs only when creating the dependency. 
```suggestion
        dependencies {
            provide<WorkExperience> { WorkExperience(this.resolve()) }
            provide<PaidWork> { PaidWork(this.resolve()) }
            provide<List<PaidWork>> { listOf(this.resolve()) }
        }

        assertFailsWith<CircularDependencyException> {
            dependencies.create<WorkExperience>()
        }
```